client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/JavaXConstraintHandler.java

```diff
@@ -42,170 +46,186 @@ public static void handleParam(NamedTypedValue namedTypedValue, Annotation annot
-                size[1] = (Integer) annotation.annotationType().getDeclaredMethod("max").invoke(annotation);
-                return size;
-            } catch (NoSuchMethodException | InvocationTargetException |IllegalAccessException e) {
-                throw new RuntimeException("ERROR: fail to process size");
-            }
+        String pattern = null;
+        try {
+            pattern = (String) annotation.annotationType().getDeclaredMethod("regexp").invoke(annotation);
+        } catch (NoSuchMethodException | InvocationTargetException |IllegalAccessException e) {
+            throw new RuntimeException("ERROR: fail to process regexp");
+        }
 
+        if (pattern == null){
+            SimpleLogger.error("ERROR: Pattern regexp is null");
+            return false;
         }
-        return null;
+
+        if (namedTypedValue instanceof StringParam){
+            ((StringParam)namedTypedValue).setPattern(pattern);
+        }  else {
+            SimpleLogger.error("ERROR: Do not solve class "+ namedTypedValue.getType().getFullTypeName() + " with its Size");
+            return false;
+        }
+        return true;
     }
 
-    private static String handlePattern(Annotation annotation)  {
-        Class<?> cons = annotation.annotationType();
-        /*
-            based on https://docs.oracle.com/javaee/7/api/javax/validation/constraints/Pattern.html
+    private static boolean handleMax(NamedTypedValue namedTypedValue, Annotation annotation, JavaXConstraintSupportType supportType){
+
+         /*
+            based on
+            https://javaee.github.io/javaee-spec/javadocs/javax/validation/constraints/Max.html
+            https://javaee.github.io/javaee-spec/javadocs/javax/validation/constraints/DecimalMax.html
             null elements are considered valid.
          */
-        if (cons.getSimpleName().equals("Pattern")){
+        Long max = null;
+        Boolean inclusive = true;
+        try {
+            // TODO might change long to BigDecimal
+            if (supportType == JavaXConstraintSupportType.DECIMAL_MAX){
+                String maxStr =  (String) annotation.annotationType().getDeclaredMethod("value").invoke(annotation);
+                max = Long.valueOf(maxStr);
+                inclusive = (Boolean) annotation.annotationType().getDeclaredMethod("inclusive").invoke(annotation);
+            }else
+                max = (Long) annotation.annotationType().getDeclaredMethod("value").invoke(annotation);
 
-            try {
-                return (String) annotation.annotationType().getDeclaredMethod("regexp").invoke(annotation);
-            } catch (NoSuchMethodException | InvocationTargetException |IllegalAccessException e) {
-                throw new RuntimeException("ERROR: fail to process regexp");
-            }
+        } catch (NoSuchMethodException | InvocationTargetException |IllegalAccessException e) {
+            throw new RuntimeException("ERROR: fail to process max");
```

Comment:
when rethrowing an exception, add its output in the string message, or add it as a cause as last parameter in this constructor